src/PhpDoc/Tag/TypeAliasTag.php

```diff
@@ -0,0 +1,40 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\PhpDoc\Tag;
+
+class TypeAliasTag
+{
+
+	private string $alias;
+
+	private string $type;
```

Comment:
Why this isn't a Type but a string?

Reply:
Because resolving a Type here would be too early, creating a chicken-and-egg problem – to support nested aliases such as `@phpstan-type ListOfAliases AnotherAlias[]`, aliases need to be resolved only after all of them are gathered. Also, imported aliases need to be resolved in the scope of the class that declared them instead of the importing one.

Eventually, I've [wrapped type aliases](https://github.com/phpstan/phpstan-src/pull/460/commits/ee81ee75a09833a73e746e70a7b24f7866e6e6df) into a `TypeAlias` class along with the NameScope in which they are declared, and they are resolved lazily when they are needed.

Reply:
Great, thank you for the explanation :)